### PR TITLE
Merchant test build

### PR DESCRIPTION
**Pull request by kylejschulz**
Here's the nearly complete iteration 0 of the merchant and merchant repo side. We still need to integrate item and item repo and finish troubleshooting the sales_engine class as well as troubleshooting the spec harness.

### CODE REVIEW

File: lib/merchant.rb [hunk 1]
```diff
@@ -1,32 +1,14 @@
...
-  end
-
-  # def create(args)
-  #   @merchants << merchant.new({:id => highest_merchant_id_plus_one,
-  #                 :args[:name] => ?)
-  # end
-
-  def update(id, attributes)
-
+  def update(name)
```
Comment (jequeza) [inline lib/merchant.rb:10]: Good call to have this method on the merchant class itself. A merchant should be able to update its name!

File: lib/merchant_repository.rb [hunk 2]
```diff
@@ -1,48 +1,69 @@
...
+  def create(hash)
+    new_merchant = Merchant.new(highest_merchant_id_plus_one, hash[:name])
+    @merchants << new_merchant
+    new_merchant
   end
 
-  def update(id, attributes)
+  def update(id, name_hash)
+    update_merchant = find_by_id(id)
+    update_merchant.update(name_hash[:name]) if !name_hash[:name].nil?
```
Comment (jequeza) [inline lib/merchant_repository.rb:58]: Awesome logic! If the name is not nil then update it! this sure handles calling something on nil

File: lib/merchant.rb (hunk 1 above)
Comment (kylejschulz) [reply lib/merchant.rb:10]: Yeah, i didn't think it made sense to build that into the merchant_repo class. I'm still a little on the fence of whether or not it makes sense to have the update and delete methods in the repo class. Do you think that was a smart move or should we move them to merchant class?

File: lib/sales_engine.rb [hunk 3]
```diff
@@ -1,30 +1,32 @@
...
-              :merchants => merchant_path,
-            }
+# item_path = "./data/items.csv"
+# merchant_path = "./data/merchants.csv"
+# arguments = {
+#               :items     => item_path,
+#               :merchants => merchant_path,
+#             }
 # se = SalesEngine.new(arguments)
 # require "pry"; binding.pry
```
Comment (jequeza) [inline lib/sales_engine.rb:32]: May need to remove this pry to keep code clean.

File: lib/merchant.rb (hunk 1 above)
Comment (jequeza) [reply lib/merchant.rb:10]: I think that it's a good idea to have it in the merchant class. If we need to change it we can always make the move! 👍 Great work dev!

**Pull request merged**